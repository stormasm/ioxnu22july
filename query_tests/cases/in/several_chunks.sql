-- IOX_SETUP: ManyFieldsSeveralChunks

-- validate we have access to information schema for listing system tables
-- IOX_COMPARE: sorted
SELECT * from h2o;
-- Plan will look like:
--  . Two chunks (one parquet and one from ingester) neither overlap nor contain duplicate
--     --> scan in one scan node
--  . Three parquet chunks overlapped -> dedup without resort
EXPLAIN SELECT * from h2o;

-- Only selct fields and time
-- IOX_COMPARE: sorted
select temp, other_temp, time from h2o;
EXPLAIN select temp, other_temp, time from h2o;

-- early pruning
EXPLAIN SELECT * from h2o where time >= to_timestamp('1970-01-01T00:00:00.000000250+00:00');
