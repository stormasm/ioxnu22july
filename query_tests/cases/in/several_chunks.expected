-- Test Setup: ManyFieldsSeveralChunks
-- SQL: SELECT * from h2o;
-- Results After Sorting
+---------+------------+-------+------+--------------------------------+
| city    | other_temp | state | temp | time                           |
+---------+------------+-------+------+--------------------------------+
| Andover | 72.4       | CA    |      | 1970-01-01T00:00:00.000000150Z |
| Andover |            | CA    | 67.3 | 1970-01-01T00:00:00.000000500Z |
| Boston  | 68.2       | MA    |      | 1970-01-01T00:00:00.000000450Z |
| Boston  | 80         | MA    |      | 1970-01-01T00:00:00.000000250Z |
| Boston  |            | MA    | 70.4 | 1970-01-01T00:00:00.000000050Z |
| Boston  |            | MA    | 80.7 | 1970-01-01T00:00:00.000000350Z |
| Boston  |            | MA    | 88.6 | 1970-01-01T00:00:00.000000230Z |
+---------+------------+-------+------+--------------------------------+
-- SQL: EXPLAIN SELECT * from h2o;
+---------------+---------------------------------------------------------------------------------------------------------------------+
| plan_type     | plan                                                                                                                |
+---------------+---------------------------------------------------------------------------------------------------------------------+
| logical_plan  | Projection: #h2o.city, #h2o.other_temp, #h2o.state, #h2o.temp, #h2o.time                                            |
|               |   TableScan: h2o projection=[city, other_temp, state, temp, time]                                                   |
| physical_plan | ProjectionExec: expr=[city@0 as city, other_temp@1 as other_temp, state@2 as state, temp@3 as temp, time@4 as time] |
|               |   UnionExec                                                                                                         |
|               |     DeduplicateExec: [city@0 ASC,state@2 ASC,time@4 ASC]                                                            |
|               |       SortPreservingMergeExec: [city@0 ASC,state@2 ASC,time@4 ASC]                                                  |
|               |         UnionExec                                                                                                   |
|               |           IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                           |
|               |           IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                           |
|               |           IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                           |
|               |     IOxReadFilterNode: table_name=h2o, chunks=2 predicate=Predicate                                                 |
|               |                                                                                                                     |
+---------------+---------------------------------------------------------------------------------------------------------------------+
-- SQL: select temp, other_temp, time from h2o;
-- Results After Sorting
+------+------------+--------------------------------+
| temp | other_temp | time                           |
+------+------------+--------------------------------+
| 67.3 |            | 1970-01-01T00:00:00.000000500Z |
| 70.4 |            | 1970-01-01T00:00:00.000000050Z |
| 80.7 |            | 1970-01-01T00:00:00.000000350Z |
| 88.6 |            | 1970-01-01T00:00:00.000000230Z |
|      | 68.2       | 1970-01-01T00:00:00.000000450Z |
|      | 72.4       | 1970-01-01T00:00:00.000000150Z |
|      | 80         | 1970-01-01T00:00:00.000000250Z |
+------+------------+--------------------------------+
-- SQL: EXPLAIN select temp, other_temp, time from h2o;
+---------------+---------------------------------------------------------------------------------------+
| plan_type     | plan                                                                                  |
+---------------+---------------------------------------------------------------------------------------+
| logical_plan  | Projection: #h2o.temp, #h2o.other_temp, #h2o.time                                     |
|               |   TableScan: h2o projection=[other_temp, temp, time]                                  |
| physical_plan | ProjectionExec: expr=[temp@1 as temp, other_temp@0 as other_temp, time@2 as time]     |
|               |   UnionExec                                                                           |
|               |     ProjectionExec: expr=[other_temp@1 as other_temp, temp@3 as temp, time@4 as time] |
|               |       RepartitionExec: partitioning=RoundRobinBatch(4)                                |
|               |         DeduplicateExec: [city@0 ASC,state@2 ASC,time@4 ASC]                          |
|               |           SortPreservingMergeExec: [city@0 ASC,state@2 ASC,time@4 ASC]                |
|               |             UnionExec                                                                 |
|               |               IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate         |
|               |               IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate         |
|               |               IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate         |
|               |     IOxReadFilterNode: table_name=h2o, chunks=2 predicate=Predicate                   |
|               |                                                                                       |
+---------------+---------------------------------------------------------------------------------------+
-- SQL: EXPLAIN SELECT * from h2o where time >= to_timestamp('1970-01-01T00:00:00.000000250+00:00');
+---------------+------------------------------------------------------------------------------------------------------------------------------------+
| plan_type     | plan                                                                                                                               |
+---------------+------------------------------------------------------------------------------------------------------------------------------------+
| logical_plan  | Projection: #h2o.city, #h2o.other_temp, #h2o.state, #h2o.temp, #h2o.time                                                           |
|               |   Filter: #h2o.time >= TimestampNanosecond(250, None)                                                                              |
|               |     TableScan: h2o projection=[city, other_temp, state, temp, time], partial_filters=[#h2o.time >= TimestampNanosecond(250, None)] |
| physical_plan | ProjectionExec: expr=[city@0 as city, other_temp@1 as other_temp, state@2 as state, temp@3 as temp, time@4 as time]                |
|               |   CoalesceBatchesExec: target_batch_size=4096                                                                                      |
|               |     FilterExec: time@4 >= 250                                                                                                      |
|               |       RepartitionExec: partitioning=RoundRobinBatch(4)                                                                             |
|               |         UnionExec                                                                                                                  |
|               |           DeduplicateExec: [city@0 ASC,state@2 ASC,time@4 ASC]                                                                     |
|               |             SortPreservingMergeExec: [city@0 ASC,state@2 ASC,time@4 ASC]                                                           |
|               |               UnionExec                                                                                                            |
|               |                 IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate exprs: [#time >= TimestampNanosecond(250, None)]   |
|               |                 IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate exprs: [#time >= TimestampNanosecond(250, None)]   |
|               |           IOxReadFilterNode: table_name=h2o, chunks=2 predicate=Predicate exprs: [#time >= TimestampNanosecond(250, None)]         |
|               |                                                                                                                                    |
+---------------+------------------------------------------------------------------------------------------------------------------------------------+
